### PR TITLE
fix: Update autocomplete to alpha.41

### PR DESCRIPTION
**Pull request by sbellone**
### Rendering

Autocomplete rendering system has changed in alpha.39: https://github.com/algolia/autocomplete.js/pull/381
So I've updated our template rendering using the new API (using `createElement` and `Fragment`) as shown in the examples added here: https://github.com/algolia/autocomplete.js/pull/409/files

### CSS

Rendering on Webkit mobile has been broken since we've updated to alpha.38, because the Touch support introduced a few new CSS classes: https://github.com/algolia/autocomplete.js/pull/379. It has been reported by some users: https://github.com/algolia/algoliasearch-netlify/issues/197 and [here](https://discourse.algolia.com/t/algolia-netlify-frontend-problem-on-responsive/11907/5).

It was difficult to cherry pick the relevant changes into our own CSS.
So to ease future updates, I've re-imported the whole `autocomplete-theme-classic`'s CSS in its own file, and imported it as-is into our `index.scss`, which now only contains the overridden properties.

### Misc
An other thing that has changed is the way to highlight nested attributes, we now need to pass an array to `highlightHit()`: https://github.com/algolia/autocomplete.js/pull/418

### CODE REVIEW

File: frontend/src/index.scss [hunk 1]
```diff
@@ -1,267 +1,94 @@
...
+:root {
   --color-mark: #{$color-mark};
-  --color-background: #{$color-bg};
-  --color-selected: #{$color-bg-selected};
-  --color-text: #{$color-text};
-  --color-input-icon: #{$color-input-icon};
-  --color-source-icon: #{$color-source-icon};
-  --height: #{$height};
-  --height-icon: #{$height-icon};
-  --font-size: #{$font-size};
```
Comment (bodinsamuel) [inline frontend/src/index.scss:32]: CSS variable are part of the exposed API, I would prefer to keep them (even if it's the same output than `@import 'autocomplete-theme-classic';`

**Review by bodinsamuel — COMMENTED** [review]
There are clear difference of design between prod and this PR, I think we should be able to merge this PR without breaking anything.
ref: https://algoliasearch-netlify.netlify.app/

**Comment by sbellone** [comment]
Ok the design is now almost exactly like the current default one.

File: frontend/src/index.scss [hunk 2]
```diff
@@ -32,236 +35,97 @@ $font-size: $font-size-m;
   --font-size: #{$font-size};
```
Comment (bodinsamuel) [inline frontend/src/index.scss:35]: Still unused:
```
--color-background
--color-selected
--color-text
--color-source-icon
--font-size
```

**Comment by bodinsamuel** [comment]
I don't have the logo anymore, can you double check? 
And a unwanted padding (top/bottom) on the panel

**Comment by sbellone** [comment]
Good catch, it was removed by mistake in alpha40, fixed in alpha41.

**Review by bodinsamuel — APPROVED** [review]
LGTM

**Pull request merged**